mne/io/base.py

```diff
@@ -836,6 +836,72 @@ def __setitem__(self, item, value):
         # set the data
         self._data[sel, start:stop] = value
 
+    def _get_data(self, sel, start, stop):
+        """Get data omitting bad segments."""
+        if self.annotations is None:
+            return self[sel, start:stop]
+        sfreq = self.info['sfreq']
+        bads = [idx for idx, desc in enumerate(self.annotations.description)
+                if desc.upper().startswith('BAD')]
+        onsets = self.annotations.onset[bads]
+        onsets = _onset_to_seconds(self, onsets)
```

Comment:
Why go to seconds, and then back to samples? Everything else is done in samples. It would be easier just to convert `duration` to samples, no?